CryptoPrices/Sources/MyCoin/Sources/MyCoin/ChartSectionView/PriceLineChartView.swift

```diff
@@ -61,6 +60,13 @@ struct PriceLineChartView: UIViewRepresentable {
         }
         dataSet.drawFilledEnabled = true
     }
+
+    private func formatNoDataText(uiView: LineChartView) {
+        uiView.noDataText = Strings.MyCoin.PriceLineChart.noDataText
+        uiView.noDataFont = Fonts.Inter.medium.font(size: 14.0)
+        uiView.noDataTextColor = Colors.textMedium.color
+        uiView.noDataTextAlignment = .center
+    }
```

Comment:
Regarding how we implement `PriceLineChartView` which conforms to `UIViewRepresentable`, I have some suggestions. Please consider checking this, and hope you find it helpful 🙏 

According to [Apple documentation](https://developer.apple.com/documentation/swiftui/uiviewrepresentable/makeuiview(context:)):
> The system calls this method [[makeUIView(context:)](https://developer.apple.com/documentation/swiftui/uiviewrepresentable/makeuiview(context:))] only once, when it creates your view for the first time. For all subsequent updates, the system calls the [updateUIView(_:context:)](https://developer.apple.com/documentation/swiftui/uiviewrepresentable/updateuiview(_:context:)) method.

So there're some points worth noting:
- We don't have to keep an instance of `LineChartView` around as a stored property (`let lineChart = LineChartView()`. We just create one and return it in the `makeUIView(context:)`. When SwiftUI wants to make an update, it hands us this exact instance as the parameter in `updateUIView(_ uiView: LineChartView, context: Context)`.
- Any view configurations are meant to be set once, we should place them in the `makeUIView(context:)` together with creating the view. Any view configurations that require change according to the latest state and data should be placed in the `updateUIView(_:context:)` while making the view update.
  
  The exception I believe is that if the one-shot configuration (set once) is expensive with maybe a heavy computation, we can delay by placing it in the `updateUIView(_:context:)` because we want to return the view as quickly as possible for the users. The SwiftUI always calls this method at least once right after `makeUIView(context:)` even if the initial state doesn't change. If that'd be a case, we're still able to set it once by simply adding a `Bool` flag.

Therefore, the _core definition_ of `PriceLineChartView` can be rewritten with small adjustments and arrangements as follows 🙏 
```swift
struct PriceLineChartView: UIViewRepresentable {

    let entries: [ChartDataEntry]

    func makeUIView(context: Context) -> LineChartView {
        let lineChart = LineChartView()

        lineChart.rightAxis.enabled = false
        lineChart.leftAxis.enabled = false
        lineChart.xAxis.enabled = false
        lineChart.legend.enabled = false
        lineChart.minOffset = 0.0
        lineChart.isUserInteractionEnabled = false
        lineChart.noDataText = Strings.MyCoin.PriceLineChart.noDataText
        lineChart.noDataFont = Fonts.Inter.medium.font(size: 14.0)
        lineChart.noDataTextColor = Colors.textMedium.color
        lineChart.noDataTextAlignment = .center
        lineChart.delegate = context.coordinator

        return lineChart
    }

    func updateUIView(_ uiView: LineChartView, context: Context) {
        defer {
            uiView.notifyDataSetChanged()
        }

        guard !entries.isEmpty else {
            uiView.data = nil
            return
        }

        let dataSet = LineChartDataSet(entries: entries)
        formatDataSet(dataSet: dataSet)
        uiView.data = LineChartData(dataSet: dataSet)
    }

    // other methods
    // ...
}
```

Reply:
Your suggestion is definitely a good improvement, thanks for the explanation! I will update it 🚀

Reply:
Regarding the guard check, it might not really be necessary and `if else` check should be cleaner here instead 🙏